x/fungible/keeper/msg_server_update_system_contract_test.go

```diff
@@ -74,6 +75,41 @@ func TestKeeper_UpdateSystemContract(t *testing.T) {
 		require.Equal(t, newSystemContract.Hex(), queryZRC20SystemContract(gas2))
 	})
 
+	t.Run("can update the system contract if system contract not found", func(t *testing.T) {
+		k, ctx, _, zk := keepertest.FungibleKeeper(t)
+		msgServer := keeper.NewMsgServerImpl(*k)
+		k.GetAuthKeeper().GetModuleAccount(ctx, types.ModuleName)
+		admin := sample.AccAddress()
+		setAdminPolicies(ctx, zk, admin, observertypes.Policy_Type_group2)
+
+		chains := zetacommon.DefaultChainsList()
+		require.True(t, len(chains) > 1)
+		require.NotNil(t, chains[0])
+		require.NotNil(t, chains[1])
+
+		wzeta, err := k.DeployWZETA(ctx)
+		require.NoError(t, err)
+
+		factory, err := k.DeployUniswapV2Factory(ctx)
+		require.NoError(t, err)
+
+		router, err := k.DeployUniswapV2Router02(ctx, factory, wzeta)
+		require.NoError(t, err)
+
+		// deploy a new system contracts
+		newSystemContract, err := k.DeployContract(ctx, systemcontract.SystemContractMetaData, wzeta, factory, router)
+		require.NoError(t, err)
+
+		// can update the system contract
+		_, err = msgServer.UpdateSystemContract(ctx, types.NewMsgUpdateSystemContract(admin, newSystemContract.Hex()))
+		require.NoError(t, err)
+
+		// can retrieve the system contract
+		sc, found := k.GetSystemContract(ctx)
+		require.True(t, found)
+		require.Equal(t, newSystemContract.Hex(), sc.SystemContract)
+	})
```

Comment:
Maybe we can deploy another system contract and check the previous one can be overwritten